src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BankHolidayService.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.entity.BankHolidayCache
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.external.BankHolidays
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.repository.BankHolidayCacheRepository
+import java.time.LocalDate
+import java.time.LocalDateTime
+import javax.transaction.Transactional
+
+@Service
+class BankHolidayService(
+  private val bankHolidayApiClient: BankHolidayApiClient,
+  private val bankHolidayCacheRepository: BankHolidayCacheRepository,
+  private val objectMapper: ObjectMapper
+) {
+
+  @Transactional
+  fun refreshCache() {
+    var cached = bankHolidayCacheRepository.findFirstByOrderByIdAsc()
+    if (cached == null || cached.cachedAt.toLocalDate().isBefore(LocalDate.now())) {
```

Comment:
Not sure what the purpose of this if condition is?

Reply:
So the idea is that the schedule will run every 5mins for an hour in the evening. The re-running is important in case the request fails. 

This condition checks if the cache is empty, or if the cache has not yet been updated today.